cider/exec/template/DateTimeIR.cpp

```diff
@@ -224,6 +264,42 @@ llvm::Value* CodeGenerator::codegen(const Analyzer::DatediffExpr* datediff_expr,
       datediff_fname, get_int_type(64, cgen_state_->context_), datediff_args);
 }
 
+std::unique_ptr<CodegenColValues> CodeGenerator::codegenDateDiff(
+    const Analyzer::DatediffExpr* datediff_expr,
+    const CompilationOptions& co) {
+  AUTOMATIC_IR_METADATA(cgen_state_);
+
+  auto start = codegen(datediff_expr->get_start_expr(), co, true);
+  auto start_fixsize = dynamic_cast<FixedSizeColValues*>(start.get());
+  auto start_value = start_fixsize->getValue();
+  CHECK(start_value->getType()->isIntegerTy(64));
+
+  auto end = codegen(datediff_expr->get_end_expr(), co, true);
+  auto end_fixsize = dynamic_cast<FixedSizeColValues*>(end.get());
+  auto end_value = end_fixsize->getValue();
+  CHECK(start_value->getType()->isIntegerTy(64));
+
+  const auto& start_ti = datediff_expr->get_start_expr()->get_type_info();
+  const auto& end_ti = datediff_expr->get_end_expr()->get_type_info();
+
+  std::vector<llvm::Value*> datediff_args{
+      cgen_state_->llInt(static_cast<int32_t>(datediff_expr->get_field())),
+      start_value,
+      end_value};
+  std::string datediff_fname{"DateDiff"};
+  if (start_ti.is_high_precision_timestamp() || end_ti.is_high_precision_timestamp()) {
+    datediff_fname += "HighPrecision";
+    datediff_args.push_back(
+        cgen_state_->llInt(static_cast<int32_t>(start_ti.get_dimension())));
+    datediff_args.push_back(
+        cgen_state_->llInt(static_cast<int32_t>(end_ti.get_dimension())));
+  }
+  llvm::Value* value = cgen_state_->emitExternalCall(
+      datediff_fname, get_int_type(64, cgen_state_->context_), datediff_args);
+
+  return std::make_unique<FixedSizeColValues>(value, start_fixsize->getNull());
```

Comment:
should the null be determined like `start_fixsize->getNull() || end_fixsize->getNull()`?

Reply:
yes, it should be determined  by both parties， but the logic will be little complex, I fixed it in recent update.